### PR TITLE
Fix addIntention

### PR DESCRIPTION
**Pull request by rachelcarmena**
There are 2 `addIntention` functions. This PR renames one of them to `registerIntention` to follow the _pattern_:

```
  /**
   * registers [intention]
   */
  fun IdeMetaPlugin.registerIntention(
    intention: IntentionAction
  ): ExtensionPhase =
    IntentionExtensionProvider.RegisterIntention(intention)

  /**
   * unregisters [intention] from the editor
   */
  fun IdeMetaPlugin.unregisterIntention(
    intention: IntentionAction
  ): ExtensionPhase =
    IntentionExtensionProvider.UnregisterIntention(intention)
```

Besides it adds types for `f` and `editor` because they cannot be inferred.

### CODE REVIEW

**Comment by i-walker** [comment]
`addntention` should not be changed it is the naming convention we follow throughout the Ide.

**Comment by rachelcarmena** [comment]
Thanks for the review, @danimontoya , @JorgeCastilloPrz :raised_hands: 

@i-walker , thanks!! 

> `addntention` should not be changed it is the naming convention we follow throughout the Ide.

There are 2 `IdeMetaPlugin.addIntention` in the [original file](https://github.com/arrow-kt/arrow-meta/blob/master/idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt). I'm just changing the first one.

I'll wait for your review again, @i-walker :+1: Thanks!! :raised_hands:

File: idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ interface IntentionSyntax : IntentionUtilitySyntax {
   /**
    * registers [intention]
    */
-  fun IdeMetaPlugin.addIntention(
+  fun IdeMetaPlugin.registerIntention(
```
Comment (i-walker) [inline idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt:31]:
```suggestion
  fun IdeMetaPlugin.addIntention(
```

**Comment by danimontoya** [comment]
if this helps.. we've done the same for folding builder case:
```
  fun IdeMetaPlugin.addFoldingBuilder(
    placeHolderText: (node: ASTNode) -> String?,
    foldRegions: (element: PsiElement, document: Document, quick: Boolean) -> List<FoldingDescriptor>,
    isCollapsedByDefault: (node: ASTNode) -> Boolean,
    lang: Language = KotlinLanguage.INSTANCE): ExtensionPhase =
    registerFoldingBuilder(foldingBuilder(placeHolderText, foldRegions, isCollapsedByDefault), lang)

  fun IdeMetaPlugin.registerFoldingBuilder(foldingBuilder: FoldingBuilder, lang: Language = KotlinLanguage.INSTANCE): ExtensionPhase =
    ExtensionProvider.AddFoldingBuilder(lang, foldingBuilder)
```
Let me know if I need to change it

**Comment by i-walker** [comment]
Yes Dani, I overlooked that :)

File: idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt [hunk 2]
```diff
@@ -80,7 +81,7 @@ interface IntentionSyntax : IntentionUtilitySyntax {
     applyTo: KtPsiFactory.(element: K, editor: Editor) -> Unit = Noop.effect3,
     priority: PriorityAction.Priority = PriorityAction.Priority.LOW
   ): ExtensionPhase =
-    addIntention(ktIntention(text, kClass, isApplicableTo, applyTo, priority))
+    registerIntention(ktIntention(text, kClass, isApplicableTo, applyTo, priority))
```
Comment (i-walker) [inline idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt:84]:
```suggestion
    addIntention(ktIntention(text, kClass, isApplicableTo, applyTo, priority))
```

**Pull request merged**